### PR TITLE
Fixed issue with casting

### PR DESCRIPTION
**Pull request by vaslabs**
Hello,

I can't apologise enough for this but this fixes an issue with the region on the pull request I had earlier. I just realised by upgrading to this version from the version I uploaded to our artifactory.

The problem was in withRegion that does a cast and it throws a class cast exception. 

The setRegion works. Again, apologies, I shouldn't have changed it last thing or at least I should have run the tests we have again.

Thanks,
Vasilis

### CODE REVIEW

File: client/src/main/scala/com/gilt/gfc/aws/kinesis/client/KinesisPublisher.scala [hunk 1]
```diff
@@ -231,7 +231,7 @@ class KinesisPublisherImpl (
     val conf = cf.getConfig.withRetryPolicy(rp)
 
     val client = new AmazonKinesisClient(awsCredentialsProvider, conf)
-    awsRegion.fold(client)(region => client.withRegion(region))
+    awsRegion.fold(client)(region => {client.setRegion(region); client})
```
Comment (andreyk0) [inline client/src/main/scala/com/gilt/gfc/aws/kinesis/client/KinesisPublisher.scala]: Ah, thanks for correcting, I didn't notice.

Mind changing it to something like
```
awsRegion.foreach { region => client.setRegion(region }
client
```

**Comment by andreyk0** [comment]
Never mind, I've applied a slight variation of your fix to master (want to re-release quickly, before anyone downloads affected version)

Here's the commit, please let me know if you see any other issues with this.


```
commit 33684b978c43b7e6224ef9e24a6325d3c66951cf
Author: Andrey Kartashov <akartashov@gilt.com>
Date:   Tue Mar 7 13:59:32 2017 -0500

    Fix for class cast exception.
    
    A slight adaptation of https://github.com/gilt/gfc-aws-kinesis/pull/22

diff --git a/client/src/main/scala/com/gilt/gfc/aws/kinesis/client/KinesisPublisher.scala b/client/src/main/scala/com/gilt/gfc/aws/kinesis/client/KinesisPublisher.scala
index 6e620db..6039348 100644
--- a/client/src/main/scala/com/gilt/gfc/aws/kinesis/client/KinesisPublisher.scala
+++ b/client/src/main/scala/com/gilt/gfc/aws/kinesis/client/KinesisPublisher.scala
@@ -231,7 +231,8 @@ class KinesisPublisherImpl (
     val conf = cf.getConfig.withRetryPolicy(rp)
 
     val client = new AmazonKinesisClient(awsCredentialsProvider, conf)
-    awsRegion.fold(client)(region => client.withRegion(region))
+    awsRegion.foreach(region => client.setRegion(region))
+    client
   }
 
 ```

**Comment by vaslabs** [comment]
Thanks. I'll up-version and test again tomorrow and let you know. Should be ok.

**Comment by vaslabs** [comment]
Works as expected.

**Pull request closed without merging**